### PR TITLE
Fix bug for get index state

### PR DESCRIPTION
**Pull request by xiaocai2333**
Signed-off-by: cai.zhang <cai.zhang@zilliz.com>
issue: #21269 
/kind bug

### CODE REVIEW

File: internal/datacoord/index_service.go [hunk 1]
```diff
@@ -355,17 +355,18 @@ func (s *Server) completeIndexInfo(indexInfo *indexpb.IndexInfo, index *model.In
...
 	case cntFailed > 0:
 		indexInfo.State = commonpb.IndexState_Failed
 		indexInfo.IndexStateFailReason = failReason
-	case cntFinished == allCnt:
-		indexInfo.State = commonpb.IndexState_Finished
-	default:
+	case cntInProgress > 0 || cntUnissued > 0:
 		indexInfo.State = commonpb.IndexState_InProgress
+	case cntNone > 0:
+		indexInfo.State = commonpb.IndexState_IndexStateNone
```
Comment (czs007) [inline internal/datacoord/index_service.go:367]: It is recommended to add a Warn log, if cntNone is greater than 0

**Comment by xiaocai2333** [comment]
rerun ut

**Pull request merged**